### PR TITLE
Upgrade to mypy 1.7.0

### PR DESCRIPTION
**Pull request by intgr**
## Related issues

* #1828

### CODE REVIEW

File: django-stubs/template/response.pyi [hunk 1]
```diff
@@ -19,7 +19,7 @@ class ContentNotRenderedError(Exception): ...
 class SimpleTemplateResponse(HttpResponse):
     content: Any
     closed: bool
-    cookies: SimpleCookie[str]
+    cookies: SimpleCookie[str]  # type: ignore[type-arg]
```
Comment (intgr) [inline django-stubs/template/response.pyi:22]: These will be fixed in #1702.

File: requirements.txt [hunk 2]
```diff
@@ -12,4 +12,4 @@ Django==4.2.7
 -e .[compatible-mypy]
 
 # Overrides:
-mypy==1.6.1
+git+https://github.com/python/mypy@release-1.7
```
Comment (intgr) [inline requirements.txt]: REMINDER update to `mypy==1.7.0` once released.

Comment (intgr) [reply requirements.txt]:
```suggestion
mypy==1.7.0
```

**Review by sobolevn — APPROVED** [review]
Nice! Thanks!

**Pull request merged**